homeassistant/components/xiaomi/camera.py

```diff
@@ -133,14 +136,20 @@ def get_latest_video_url(self):
             video = videos[-1]
 
         return "ftp://{0}:{1}@{2}:{3}{4}/{5}".format(
-            self.user, self.passwd, self.host, self.port, ftp.pwd(), video
+            self.user, self.passwd, host, self.port, ftp.pwd(), video
         )
 
     async def async_camera_image(self):
         """Return a still image response from the camera."""
         from haffmpeg.tools import ImageFrame, IMAGE_JPEG
 
-        url = await self.hass.async_add_job(self.get_latest_video_url)
+        try:
+            host = self.host.async_render()
+        except TemplateError as exc:
+            _LOGGER.error('Error parsing template %s: %s', self.host, exc)
+            return self._last_image
+
+        url = await self.hass.async_add_job(self.get_latest_video_url, host)
```

Comment:
Please use `self.hass.async_add_executor_job`.

Reply:
you mean to change self.hass.async_add_job to self.hass.async_add_executor_job ?

Reply:
Yes.